File: Dataset/cs/Fileprint.cs

//  Fileprint.cs  (c) 2003 Kari Laitinen

//  www.naturalprogramming.com

using System ;
using System.IO ;

class Fileprint
{
   static void Main()
   {
      Console.Write( "\n This program prints the contents of a text"
                   + "\n file to the screen. Give a file name: " ) ;

      string  file_name_from_user  =  Console.ReadLine() ;

      try
      {
         StreamReader file_to_print  =  new  StreamReader(
                                               file_name_from_user ) ;
         int  line_counter  =  0 ;
         bool end_of_file_encountered  =  false ;

         while ( end_of_file_encountered  ==  false )
         {
            string  text_line_from_file   =  file_to_print.ReadLine() ;

            if ( text_line_from_file  ==  null )
            {
               end_of_file_encountered  =  true ;
            }
            else
            {
               Console.Write( text_line_from_file  +  "\n" ) ;
               line_counter  ++  ;
            }
         }

         file_to_print.Close() ;
         Console.Write( "\n  " +  line_counter  + " lines printed." ) ;
      }
      catch ( FileNotFoundException )
      {
         Console.Write( "\n Cannot open file " + file_name_from_user ) ;
      }
   }
}

File: Dataset/cpp/fileprint.cpp

//  fileprint.cpp  (c) 1998-2002 Kari Laitinen

#include  <iostream.h>
#include  <fstream.h>  // File handling classes.

int main()
{
   ifstream  file_to_print ;

   char   file_name_from_user[ 50 ] ;
   char   text_line_from_file[ 150 ] ;

   cout << "\n This program prints the contents of a text"
        << "\n file to the screen. Give a file name: " ;

   cin.getline( file_name_from_user,
                sizeof( file_name_from_user ) ) ;

   file_to_print.open( file_name_from_user ) ;

   if ( file_to_print.fail() )
   {
      cout << "\n Cannot open file " << file_name_from_user ;
   }
   else
   {
      int line_counter  =  0 ;

      while ( ! file_to_print.eof() )
      {
         file_to_print.getline( text_line_from_file,
                                sizeof( text_line_from_file ) ) ;

         cout  <<  text_line_from_file  <<  "\n" ;

         line_counter  ++  ;
      }

      file_to_print.close() ;

      cout  << "\n  " <<  line_counter  << " lines printed." ;
   }
}


